benchmarks/apps/babelstream/babelstream.py

```diff
@@ -0,0 +1,220 @@
+#
+# SPDX-License-Identifier: BSD-3-Clause
+import os
+import os.path as path
+import sys
+import reframe as rfm
+import reframe.utility.sanity as sn
+from reframe.core.backends import getlauncher
+# Add top-level directory to `sys.path` so we can easily import extra modules
+# from any directory.
+sys.path.append(path.join(path.dirname(__file__), '..', '..'))
+# `SpackTest` is a class for benchmarks which will use Spack as build system.
+# The only requirement is to inherit this class and set the `spack_spec`
+# attribute.
+from modules.utils import SpackTest
+
+
+
+# spack install --add babelstream%gcc@9.2.0 +thrust backend=cuda cuda_arch=70
+@rfm.simple_test
+class BabelstreamBenchmarkBase(SpackTest):
+    descr = 'Build BabelStream with Spack Build System'
+    valid_systems = ['*']
+    valid_prog_environs = ['default'] 
+    
+
+    @run_before('run')
+    def replace_launcher(self):
+        # For this benchmark we don't use MPI at all, so we always force the
+        # local launcher:
+        # <https://reframe-hpc.readthedocs.io/en/v4.1.3/tutorial_advanced.html#replacing-the-parallel-launcher>.
+        self.job.launcher = getlauncher('local')()
+    
+    @run_after('setup')
+    def setup_variables(self):
+        # Request the GPU resources necessary to run this job.
+        self.extra_resources['gpu'] = {'num_gpus_per_node': self.num_gpus_per_node}
+
+    @run_before('sanity')
+    def set_sanity_patterns(self):
+        self.sanity_patterns = sn.assert_found('Running kernels 100 times', self.stdout)
+
+    #--------------------------------------------------
+    #-- Figure of Merit Export
+    #--------------------------------------------------
+
+    # Python's built-in float type has double precision (it's a C double in CPython, a Java double in Jython).
+    # If you need more precision, get NumPy and use its numpy.float128.
+    @performance_function('MBytes/sec', perf_key='Copy')
+    def extract_copy_perf(self):
+        return sn.extractsingle(r'Copy \s+(\S+)\s+.', self.stdout, 1, float)
+
+    @performance_function('MBytes/sec', perf_key='Mul')
+    def extract_scale_perf(self):
+        return sn.extractsingle(r'Mul \s+(\S+)\s+..', self.stdout, 1, float)
+
+    @performance_function('MBytes/sec', perf_key='Add')
+    def extract_add_perf(self):
+        return sn.extractsingle(r'Add \s+(\S+)\s+.?', self.stdout, 1, float)
+
+    @performance_function('MBytes/sec', perf_key='Triad')
+    def extract_triad_perf(self):
+        return sn.extractsingle(r'Triad \s+(\S+)\s+.', self.stdout, 1, float)
+    @performance_function('MBytes/sec', perf_key='Dot')
+    def extract_dot_perf(self):
+        return sn.extractsingle(r'Dot \s+(\S+)\s+.', self.stdout, 1, float)
+
+#--------------------------------------------------
+#-- ACC
+#--------------------------------------------------
+@rfm.simple_test
+class ACCBenchmark_CPU(BabelstreamBenchmarkBase):
+    valid_systems = ['-gpu']
```

Comment:
Uhm, I just realised this will not work very well on Myriad where we have a single partition for CPU only and GPU jobs.  I'll need to think how to deal with this, I see two options:

* have two partitions for Myriad, but then in all CPU benchmarks we'll likely want to have `valid_systems = ['-gpu']`
* have a `cpu` feature for partitions

I like the latter option less, because in a sense pretty much all partitions will have CPUs, but the former option requires setting `valid_systems` explicitly everywhere.

Reply:
What if we remove the `gpu` from the features of Myriad 
https://github.com/ukri-excalibur/excalibur-tests/blob/4380722d0233438a615e0d64837339970546e712/benchmarks/reframe_config.py#L129
Or, 
Can we create another virtual `MyriadGPU` system in reframe config ?

Reply:
> What if we remove the gpu from the features of Myriad

But it does have GPUs, they can be optionally requested.

>  Can we create another virtual MyriadGPU system in reframe config ?

That was my first option :slightly_smiling_face:

Reply:
Apologies :laughing: , yes I agree with the first option!

Reply:
For the record, the first option had been implemented in #164.